src/lib/ffi/ffi_oid.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+* (C) 2017 Jack Lloyd
+* (C) 2025 Dominik Schricker
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/ffi.h>
+
+#include <botan/pk_keys.h>
+#include <botan/internal/ffi_oid.h>
+#include <botan/internal/ffi_pkey.h>
+#include <botan/internal/ffi_util.h>
+
+extern "C" {
+
+using namespace Botan_FFI;
+
+int botan_oid_from_string(botan_asn1_oid_t* oid_obj, const char* oid_str) {
+   return ffi_guard_thunk(__func__, [=]() -> int {
+      if(oid_obj == nullptr) {
+         return BOTAN_FFI_ERROR_NULL_POINTER;
+      }
+
+      *oid_obj = nullptr;
+
+      std::unique_ptr<Botan::OID> oid = std::make_unique<Botan::OID>(oid_str);
+      if(oid && !oid->empty()) {
+         *oid_obj = new botan_asn1_oid_struct(std::move(oid));
+         return BOTAN_FFI_SUCCESS;
+      } else {
+         return BOTAN_FFI_ERROR_INVALID_INPUT;
+      }
+   });
+}
+
+// todo string view thingy
+// also todo parse RSA and other such names
+
+int botan_oid_equal(botan_asn1_oid_t a_w, botan_asn1_oid_t b_w) {
+   return BOTAN_FFI_VISIT(a_w, [=](const auto& a) -> int { return a == safe_get(b_w); });
+}
+
+int botan_oid_cmp(int* result, botan_asn1_oid_t a_w, botan_asn1_oid_t b_w) {
```

Comment:
The implementation for this is basically stolen from `botan_mp_cmp`, but one thing I did notice is that `* result` isn't checked anywhere, not in the macro, nor within the function, i.e. this can segfault. Is this intended? There are a few other functions that also behave like this, e.g. `botan_mp_num_bytes`

Reply:
`result` should be checked to not be null with the appropriate error code returned if it is. The general intention is, if a FFI function is called with a pointer value that is null when it should not be, we return the dedicated null pointer error code - any function that's missing such a check has a bug.

This is a lot stricter than the C++ API; there we might at times detect/reject a null pointer but it's not assured. FFI is in a weird place because it's commonly called from languages with effectively much weaker type safety than C++, so it goes to some efforts to make errors like this non-fatal.

Reply:
Alright, then I'll fix it for this PR's methods only for now, and in a separate PR I can maybe go through `ffi.h` and fix the rest.